src/pool/srv_target.c

```diff
@@ -1060,43 +1060,70 @@ ds_pool_cont_iter(daos_handle_t ph, pool_iter_cb_t callback, void *arg)
 	return rc;
 }
 
-struct obj_iter_arg {
-	cont_iter_cb_t	callback;
+struct cont_iter_arg {
+	cont_iter_cb_t	 callback;
+	uuid_t		 po_uuid;
 	void		*arg;
+	uint32_t	 version;
+	uint32_t	 intent;
 };
 
 static int
-cont_obj_iter_cb(uuid_t cont_uuid, daos_unit_oid_t oid, daos_epoch_t eph,
-		 void *data)
+cont_iter_cb(uuid_t cont_uuid, vos_iter_entry_t *ent, void *data)
 {
-	struct obj_iter_arg *arg = data;
+	struct cont_iter_arg *arg = data;
 
-	return arg->callback(cont_uuid, oid, eph, arg->arg);
+	return arg->callback(cont_uuid, ent, arg->arg);
 }
 
 static int
-pool_obj_iter_cb(daos_handle_t ph, uuid_t co_uuid, void *data)
+pool_iter_cb(daos_handle_t ph, uuid_t co_uuid, void *data)
 {
-	return ds_cont_obj_iter(ph, co_uuid, cont_obj_iter_cb, data);
+	struct cont_iter_arg *arg = data;
+
+	switch (arg->intent) {
+	case DAOS_INTENT_REBUILD: {
+		int	rc;
+
+		/* For rebuild case, we need to resync DTXs' status firstly. */
+		rc = dtx_resync(ph, arg->po_uuid, co_uuid, arg->version);
+		if (rc != 0) {
```

Comment:
This is what we've called "force commit" before scan in our discussion, right?

Reply:
Yes, it is the DTX resync before object rebuild.